### PR TITLE
Add optional user `display name` within community

### PR DESCRIPTION
**Pull request by megrogan**
I'll integrate locally with the website first to be sure we're not missing anything on the backend

### CODE REVIEW

File: backend/canisters/community/impl/src/model/channels.rs [hunk 1]
```diff
@@ -261,6 +271,15 @@ impl Channel {
...
         let updates_from_events = chat.summary_updates_from_events(since, user_id, now);
 
+        let latest_message = can_view_latest_message
+            .then_some(updates_from_events.latest_message)
+            .flatten();
+
+        let latest_message_sender_display_name = latest_message
+            .as_ref()
+            .and_then(|m| community_members.get_by_user_id(&m.event.sender))
+            .and_then(|m| m.display_name.as_ref().cloned());
```
Comment (hpeebles) [inline backend/canisters/community/impl/src/model/channels.rs:281]: .as_ref().cloned()... surely just .cloned() will work?

File: backend/canisters/community/impl/src/model/channels.rs [hunk 2]
```diff
@@ -185,6 +188,11 @@ impl Channel {
         let latest_event_index = main_events_reader.latest_event_index().unwrap_or_default();
         let latest_message = if can_view_latest_message { main_events_reader.latest_message_event(user_id) } else { None };
 
+        let latest_message_sender_display_name = latest_message
+            .as_ref()
+            .and_then(|m| community_members.get_by_user_id(&m.event.sender))
+            .and_then(|m| m.display_name.as_ref().cloned());
```
Comment (hpeebles) [inline backend/canisters/community/impl/src/model/channels.rs:194]: Same here

File: backend/canisters/community/impl/src/queries/selected_updates.rs [hunk 3]
```diff
@@ -27,11 +27,17 @@ fn selected_updates_impl(args: Args, state: &RuntimeState) -> Response {
     let invited_users_last_updated = data.invited_users.last_updated();
     let events_last_updated = data.events.latest_event_timestamp();
     let user_groups_last_updated = data.members.user_groups_last_updated();
-
-    let last_updated = [invited_users_last_updated, events_last_updated, user_groups_last_updated]
-        .into_iter()
-        .max()
-        .unwrap();
+    let display_names_last_updated = data.members.display_names_last_updated();
```
Comment (hpeebles) [inline backend/canisters/community/impl/src/queries/selected_updates.rs:30]: Iterating over every member doesn't seem right. We should maintain a `last_updated` field instead.

File: backend/canisters/community/impl/src/model/channels.rs (hunk 1 above)
Comment (megrogan) [reply backend/canisters/community/impl/src/model/channels.rs:281]: Nope

File: backend/canisters/community/impl/src/model/channels.rs (hunk 2 above)
Comment (megrogan) [reply backend/canisters/community/impl/src/model/channels.rs:194]: Nope

File: backend/canisters/community/impl/src/model/channels.rs [hunk 4]
```diff
@@ -11,6 +11,8 @@ use types::{
     MAX_THREADS_IN_SUMMARY,
 };
 
+use super::members::CommunityMembers;
```
Comment (hpeebles) [inline backend/canisters/community/impl/src/model/channels.rs:14]: This shouldn't be separate from the other `use`s

**Review by hpeebles — APPROVED** [review]

**Pull request merged**